### ISSUE TITLE
YetiRank hidden dataset requirements (No docs)

### ISSUE THREAD

**Issue by wingrime** [catboost/catboost#367]
YetiRank have some "not oblivious" requirements that lack any descriptions in docs. C++ dig requires.

Train/validation sets seems should be grouped by group_ids.

Some checks lay in catboost/catboost/libs/train_lib/preprocess.cpp

It also not oblivious how make proper dataset split. `train_test_split` not enough here.

**Comment by Noxoomo** [comment]
Hi!

Some description could be found here https://tech.yandex.com/catboost/doc/dg/concepts/loss-functions-docpage/#loss-functions__ranking

And we'll add a tutorial for ranking later

> It also not oblivious how make proper dataset split. train_test_split not enough here.


YetiRank, as well as all other ranking metrics, requires a grouping of observations. 
Grouping is domain-specific. For example, for search engines one group is  documents from same query. For ranking metric one observation in dataset is effectively a group of documents.  
Learn/test/validate split for ranking objective should be group-wise (you should add all documents from one group to learn or test test, not some document to learn and some for test)

**Comment by wingrime** [comment]
Nice to see clear answer
